authoring-map/src/main/java/com/esri/arcgisruntime/sample/authoringamap/MainActivity.java

```diff
@@ -0,0 +1,335 @@
+                            // if both items are checked, add them
+                            if (mLayerListView.getCheckedItemCount() > 1) {
+                                progressDialog.show();
+                                removeLayers();
+                                mMap.getOperationalLayers().add(layer_array[0]);
+                                mMap.getOperationalLayers().add(layer_array[1]);
+                            } else { // if any one item is checked, add as layer
+                                if (mLayerListView.isItemChecked(0)) {
+                                    progressDialog.show();
+                                    removeLayers();
+                                    mMap.getOperationalLayers().add(layer_array[0]);
+                                } else if (mLayerListView.isItemChecked(1)) {
+                                    progressDialog.show();
+                                    removeLayers();
+                                    mMap.getOperationalLayers().add(layer_array[1]);
+                                } else {
+                                    removeLayers();
+                                }
+                            }
+                        }
+                    }
+                });
+                // if the progrees dialog is showing, dismiss it
+                mMapView.addLayerViewStateChangedListener(new LayerViewStateChangedListener() {
+                    @Override
+                    public void layerViewStateChanged(LayerViewStateChangedEvent layerViewStateChangedEvent) {
+                        if (progressDialog.isShowing()) {
+                            progressDialog.dismiss();
+                        }
+
+                    }
+                });
+                invalidateOptionsMenu();
+            }
+
+            public void onDrawerOpened(View drawerView) {
+                getSupportActionBar().setTitle(mDrawerTitle);
+                // calling onPrepareOptionsMenu() to hide action bar icons
+                invalidateOptionsMenu();
+
+            }
+        };
+
+        mDrawerToggle.setDrawerIndicatorEnabled(true);
+        mDrawerLayout.setDrawerListener(mDrawerToggle);
+
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setHomeButtonEnabled(true);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu_main, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // toggle nav drawer on selecting action bar app icon/title
+        int id = item.getItemId();
+
+        //noinspection SimplifiableIfStatement
+        if (id == R.id.action_save) {
+            oAuthBrowser();
+        }
+
+        // Activate the navigation drawer toggle
+        if (mDrawerToggle.onOptionsItemSelected(item)) {
```

Comment:
This can be replaced with `return mDrawerToggle.onOptionsItemSelected(item) || super.onOptionsItemSelected(item);`